### ISSUE TITLE
onnx2ncnn Gemm convert failed 

### ISSUE THREAD

**Issue by gglin001** [Tencent/ncnn#2427]
Gemm with  weight and bias convert failed, weight and bias are lost.

![image](https://user-images.githubusercontent.com/13805221/101748574-aee53980-3b07-11eb-917f-9f403d1f742b.png)

model file

[github_issue.zip](https://github.com/Tencent/ncnn/files/5671433/github_issue.zip)

**Comment by nihui** [comment]
Hi, how is your onnx generated?
I wrote a script to emulate gemm but cannot get the Gemm op

```python
import torch
import torch.nn as nn

class CatModel(nn.Module):
    def __init__(self):
        super(CatModel, self).__init__()
        self.w = torch.randn(256, 2)
        self.b = torch.randn(2)

    def forward(self, x):
        y = torch.matmul(x, self.w) + self.b
        return y

if __name__ == "__main__":
    net = CatModel()
    x = torch.randn(1, 16, 256)

    torch.onnx.export(net, x, "cat.onnx")
```

**Comment by gglin001** [comment]
I am using onnx==1.7.0 (opset v12)

"gemm.onnx" is generated by code:

``` python

import onnx
import onnx.helper
import onnx.checker
import numpy as np


def make_const_node(name, value):
    node_const = onnx.helper.make_node(
        'Constant',
        inputs=[],
        outputs=[name],
        value=onnx.helper.make_tensor(
            name='const_tensor',
            data_type=onnx.TensorProto.FLOAT,
            dims=value.shape,
            vals=value.flatten().astype(float),
        ),
    )
    return node_const

b = np.random.randn(256, 2).astype(np.float32)
node_b = make_const_node('b', b)

c = np.random.randn(2).astype(np.float32)
node_c = make_const_node('c', c)

node = onnx.helper.make_node(
    'Gemm',
    inputs=['x', 'b', 'c'],
    outputs=['y']
)

input_value = onnx.helper.make_tensor_value_info(
    name='x',
    elem_type=onnx.TensorProto.FLOAT,
    shape=(16, 256),
    doc_string="",
    shape_denotation=None,
)

output_value = onnx.helper.make_tensor_value_info(
    name='y',
    elem_type=onnx.TensorProto.FLOAT,
    shape=(16, 2),
    doc_string="",
    shape_denotation=None,
)

graph = onnx.helper.make_graph(
    nodes=[node_b, node_c, node],
    name='gemm',
    inputs=[input_value],
    outputs=[output_value],
    initializer=None,
    doc_string=None,
    value_info=[],
)

onnx.checker.check_graph(graph)
model = onnx.helper.make_model(graph)
onnx.save_model(model, 'gemm.onnx')

```

**Comment by nihui** [comment]
fixed in https://github.com/Tencent/ncnn/commit/b7440cad2deeadfcecbb125619f93358511df54d

You have to add the batch dim to your shape, eg (16, 2) to (1, 16, 2), in order to make onnx2ncnn happy

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
